jetson/percep/artag_detector.cpp

```diff
@@ -127,4 +128,43 @@ cv::aruco::drawDetectedMarkers(rgb, corners, ids);
 
 double TagDetector::getAngle(float xPixel, float wPixel){
     return atan((xPixel - wPixel/2)/(wPixel/2)* tan(fieldofView/2))* 180.0 /PI;
-}
\ No newline at end of file
+}
+
+void TagDetector::updateDetectedTagInfo(rover_msgs::Target *arTags, pair<Tag, Tag> &tagPair, Mat &depth_img, Mat &src){
+    struct tagPairs {
+    vector<int> id;
+    vector<int> locx;
+    vector<int> locy;
+    vector<int> buffer;
+    }; 
+    tagPairs tags;
+
+    tags.id.push_back(tagPair.first.id);
+    tags.locx.push_back(tagPair.first.loc.x);
+    tags.locy.push_back(tagPair.first.loc.y);
+    tags.id.push_back(tagPair.second.id);
+    tags.locx.push_back(tagPair.first.loc.x);
+    tags.locy.push_back(tagPair.first.loc.y);
+    tags.buffer.push_back(0);
+    tags.buffer.push_back(0);
+
+  for (uint i=0; i<2; i++)
+  {
+    if(tags.id[i] == -1){//no tag found
+        if(tags.buffer[i] <= 20){//send buffered tag until tag is found
+            ++tags.buffer[i];
+        } else {//if still no tag found, set all stats to -1
+            arTags[i].distance = -1;
+            arTags[i].bearing = -1;
+            arTags[i].id = -1;
```

Comment:
Could you make these edge cases clear in[ this ICD document](https://docs.google.com/document/d/1WzsQUxGt2jDygyt_J5W_eQBTJtCMHHeMhDp42rwpi78/edit)? Right now, afaik, this behaviour only lives in people's brains.

Reply:
I can do that